### PR TITLE
Adds a new toy: The Captain's Aid

### PR DESCRIPTION
**Pull request by FernandoJ8**
## About The Pull Request

This PR adds a brand new frivolous piece of hardware to help all of us spacefarers: The Captain's Aid
image

https://user-images.githubusercontent.com/80640114/148107556-dfc54d05-cab1-4c01-9736-1d34885df78f.mp4

This wonderful invention by our pals at Donk Co. will be available for the public at the toy vendor and the Collector's edition will be a possible mail reward and heirloom for our beloved captains.

## Why It's Good For The Game

People need to learn their directions in space so that we may truly achieve the levels of immersion the creators intendedᵀᴹ.
Also it's funny.

Credit to Pepsi for the idea:
image

## Changelog

:cl:
add: Adds the captain's aid
/:cl:

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -182,6 +183,34 @@
...
+	current_mode++
+	playsound(src, 'sound/items/screwdriver2.ogg', 50, TRUE)
+	if (current_mode <= modes.len)
+		to_chat(user, span_notice("You set \the [src] to [modes[current_mode]]"))
+	else
+		to_chat(user, span_notice("You turn off \the [src] "))
+		current_mode = 1
+	icon_state = "captainsaid_[modes[current_mode]]"
+	update_appearance(UPDATE_ICON)
+/obj/item/toy/captainsaid/collector
```
Comment (Fikou) [inline code/game/objects/items/toys.dm:211]: newline here

File: code/modules/vending/games.dm [hunk 2]
```diff
@@ -48,7 +48,8 @@
 		/obj/item/clothing/shoes/wheelys/rollerskates= 3,
 		/obj/item/melee/skateboard/hoverboard = 1,
 		/obj/item/storage/box/tail_pin = 1,
-		/obj/item/toy/intento = 3
+		/obj/item/toy/intento = 3,
+		/obj/item/toy/captainsaid = 1
```
Comment (Fikou) [inline code/modules/vending/games.dm:52]:
```suggestion
		/obj/item/toy/captainsaid = 1,
```

File: code/modules/jobs/job_types/captain.dm [hunk 3]
```diff
@@ -27,12 +27,13 @@
...
 
-	family_heirlooms = list(/obj/item/reagent_containers/food/drinks/flask/gold)
+	family_heirlooms = list(/obj/item/reagent_containers/food/drinks/flask/gold, /obj/item/toy/captainsaid/collector)
 
 	mail_goodies = list(
 		/obj/item/clothing/mask/cigarette/cigar/havana = 20,
 		/obj/item/storage/fancy/cigarettes/cigars/havana = 15,
-		/obj/item/reagent_containers/food/drinks/bottle/champagne = 10
+		/obj/item/reagent_containers/food/drinks/bottle/champagne = 10,
+		/obj/item/toy/captainsaid/collector
```
Comment (Fikou) [inline code/modules/jobs/job_types/captain.dm:36]: no assigned value?

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (FernandoJ8) [reply code/game/objects/items/toys.dm:211]: Aye.

File: code/modules/jobs/job_types/captain.dm (hunk 3 above)
Comment (FernandoJ8) [reply code/modules/jobs/job_types/captain.dm:36]: Oops.

File: code/modules/vending/games.dm (hunk 2 above)
Comment (FernandoJ8) [reply code/modules/vending/games.dm:52]: Hm? What for?

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -182,6 +183,35 @@
...
+	. += span_notice("You could swear you've been hearing advertisments for the 'soon upcoming' release of a tablet version for the better part of 3 years...")
+
+/obj/item/toy/captainsaid/attack_self(mob/living/user)
+	current_mode++
+	playsound(src, 'sound/items/screwdriver2.ogg', 50, TRUE)
+	if (current_mode <= modes.len)
+		to_chat(user, span_notice("You set \the [src] to [modes[current_mode]]"))
+	else
+		to_chat(user, span_notice("You turn off \the [src] "))
+		current_mode = 1
```
Comment (JohnFulpWillard) [inline code/game/objects/items/toys.dm:208]: these could probably be defines

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -182,6 +183,34 @@
...
+/*
+* Captain's Aid
+*/
+/obj/item/toy/captainsaid
+	name = "\improper Captain's Aid"
+	desc = "Every captain's greatest ally when exploring the vast emptiness of space, now with a color display!"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "captainsaid_off"
+	var/current_mode = 1
+	var/list/modes= list("off", "port", "starboard", "fore", "aft")
```
[the remark this answers is not in the document]
Comment (FernandoJ8) [reply code/game/objects/items/toys.dm:195]: Righty

File: code/modules/vending/games.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/vending/games.dm:52]: A comma at the end of a list is free and makes it easier to add to the list in the future

Comment (FernandoJ8) [reply code/modules/vending/games.dm:52]: Sure then

File: code/game/objects/items/toys.dm [hunk 6]
```diff
@@ -182,6 +183,48 @@
...
+
+/obj/item/toy/captainsaid/examine_more(mob/user)
+	. = ..()
+	. += span_notice("You could swear you've been hearing advertisments for the 'soon upcoming' release of a tablet version for the better part of 3 years...")
+
+/obj/item/toy/captainsaid/attack_self(mob/living/user)
+	current_mode++
+	playsound(src, 'sound/items/screwdriver2.ogg', 50, TRUE)
+	if (current_mode <= modes.len)
+		to_chat(user, span_notice("You set \the [src] to [modes[current_mode]]"))
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:215]:
```suggestion
		balloon_alert(user, "set to [current_mode]")
```

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -182,6 +183,48 @@
...
+	else
+		to_chat(user, span_notice("You turn off \the [src] "))
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:217]:
```suggestion
		balloon_alert(user, "turned off")
```

File: code/game/objects/items/toys.dm [hunk 8]
```diff
@@ -182,6 +183,48 @@
...
+	/// Current mode of the item, changed when cycling through modes
+	var/current_mode = CAPTAINSAID_MODE_OFF
+
+/obj/item/toy/captainsaid/examine_more(mob/user)
+	. = ..()
+	. += span_notice("You could swear you've been hearing advertisments for the 'soon upcoming' release of a tablet version for the better part of 3 years...")
+
+/obj/item/toy/captainsaid/attack_self(mob/living/user)
+	current_mode++
+	playsound(src, 'sound/items/screwdriver2.ogg', 50, TRUE)
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:213]:
```suggestion
	playsound(src, 'sound/items/screwdriver2.ogg', 50, vary = TRUE)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm (hunk 6 above)
Comment (FernandoJ8) [reply code/game/objects/items/toys.dm:215]: Balloon alerts, my beloved

File: code/game/objects/items/toys.dm (hunk 8 above)
Comment (FernandoJ8) [reply code/game/objects/items/toys.dm:213]: I don't think any other playsound proc writes it that way, but sure. Improves readability so it's a good precedent.

Comment (Mothblocks) [reply code/game/objects/items/toys.dm]: I'm trying to get more people to do it because random `TRUE` isn't readable

**Pull request merged**